client/client.go

```diff
@@ -347,25 +351,50 @@ func (r *NotaryRepository) Publish() error {
 		return err
 	}
 
+	// these are the tuf files we will need to update, serialized as JSON before
+	// we send anything to remote
+	update := make(map[string][]byte)
+
 	// check if our root file is nearing expiry. Resign if it is.
 	if nearExpiry(r.tufRepo.Root) || r.tufRepo.Root.Dirty {
-		if err != nil {
-			return err
-		}
 		root, err = r.tufRepo.SignRoot(data.DefaultExpires("root"))
+		updateRoot = true
+	}
+
+	if updateRoot {
```

Comment:
I wanted to :|  `updateRoot` is also set up near the top of the function - if there was an error reaching the trust server or if the trust server had no data on the repo, then we bootstrap from local repo files and also want to update the root, because we're not sure if the server has it.